src/browserServerBackend.ts

```diff
@@ -98,39 +93,67 @@ export class BrowserServerBackend implements ServerBackend {
     void this._context!.dispose().catch(logUnhandledError);
   }
 
-  private _defineContextSwitchTool(factories: FactoryList): Tool<any> {
-    const self = this;
-    return defineTool({
+  private _defineContextSwitchTool(isVSCode: boolean) {
+    const contextSwitchers: { name: string, description?: string, switch(options: any): Promise<void> }[] = [];
+    for (const factory of this._browserContextFactories) {
+      contextSwitchers.push({
+        name: factory.name,
+        description: factory.description,
+        switch: async () => {
+          await this._setContextFactory(factory);
```

Comment:
I don't like that we now mix them and have 2 ways to switch how we connect to the browser. Maybe migrate the context factories to the backend switchers as well or what is the plan?